ctk/common/src/main/java/org/codice/compliance/utils/sign/SimpleSign.java

```diff
@@ -327,36 +327,22 @@ private void validateSignatureAndSamlKey(Signature signature, SAMLKeyInfo samlKe
   /** Private Getters */
   private java.security.Signature getSignature(X509Certificate certificate, PrivateKey privateKey)
       throws SignatureException {
-    String jceSigAlgo = "SHA1withRSA";
-    if ("DSA".equalsIgnoreCase(certificate.getPublicKey().getAlgorithm())) {
-      jceSigAlgo = "SHA1withDSA";
-    }
-
     java.security.Signature signature;
     try {
+      String jceSigAlgo = String.format("sha1with%s", certificate.getPublicKey().getAlgorithm());
```

Comment:
❓ In @coyotesqrl's DDF PR, he just calls `certificate.getSigAlgName();` instead of this line. Did you test with that? Did that include the `sha` piece of the name?

Reply:
`certificate.getSigAlgName()` returns the algorithm that the CA used to sign the certificate, not the algorithm of the public key within the certificate.

Reply:
Did we decide that was incorrect? I don't remember.

Reply:
That was my understanding 🤷‍♂️. @blen-desta do you concur?

Reply:
Yes, `certificate.getSigAlgName()` returns the alg used by the CA

Reply:
So this is the main thing that needs to be fixed in the DDF PR. @coyotesqrl do you want one of us to comment on the DDF PR with this context?

Reply:
Comment, tell me what to do, paste a patch...you know, whatever _you_ guys think you should do.